unit-tests/src/test/scala/java/nio/file/attribute/UserPrincipalLookupServiceSuite.scala

```diff
@@ -0,0 +1,27 @@
+package java.nio.file.attribute
+
+import java.nio.file.FileSystems
+
+object UserPrincipalLookupServiceSuite extends tests.Suite {
+  val lookupService = FileSystems.getDefault.getUserPrincipalLookupService
+
+  test("lookupPrincipalByName succeeds for `root` user") {
+    assert(
+      lookupService.lookupPrincipalByName("root").getName == "root")
+  }
+
+  test("lookupPrincipalByName throws exception for `gobbledygook` user") {
+    assertThrows[UserPrincipalNotFoundException](
+      lookupService.lookupPrincipalByName("gobbledygook"))
+  }
+
+  test("lookupPrincipalByGroupName succeeds for `root` group") {
+    assert(
+      lookupService.lookupPrincipalByGroupName("root").getName == "root")
+  }
+
+  test("lookupPrincipalByGroupName throws exception for `gobbledygook` group") {
+    assertThrows[UserPrincipalNotFoundException](
+      lookupService.lookupPrincipalByGroupName("gobbledygook"))
+  }
+}
```

Comment:
Studying the test suite I kept asking myself:  What confidence do I/we have that the problem described
as a reproducer in the original commit is actually both fixed and shown to be fixed?

After a number of design studies, I believe a simple solution is to add a test here which does a 
lookup of a strictly numeric group name believed to not exist on Travis, say  65501 (not 65534).

This test would get a short comment that it shows the reproducer in the base commit note
is fixed.  Yes, future devos will have to spend time figuring out this is true. It is not face evident
but a comment will give some future devo a clue & starting place.

The second edit would be to edit the base commit to say that identify the test & file that gives the
essence of the long form reproducer given.  I would keep that long form raw reproducer.

[needs scalafmt]

test("lookupPrincipalByGroupName succeeds for gid only group 65501") {
 if (isTravis) {
    val expected = "65501"
    assert(
      lookupService.lookupPrincipalByGroupName(expected).getName == expected)
  }
}

Reply:
To be honest, I thought this will just not work since you cannot simply pass a UID / GID for name.

But you're right. There's a crazy [loophole](https://github.com/keerath/openjdk-8-source/blob/5f6e9d42a9f6b6736100c9c6f43f5f5ea1570cfb/jdk/src/solaris/classes/sun/nio/fs/UnixUserPrincipals.java#L150-L157) (which is not even documented), but you can create a User or Group principal if you pass an arbitrary integer value as String to the `lookupByName` or `lookupByGroupName` methods of the `UserPrincipalLookupService`. :scream:

Reply:
I think, in this light, we should add tests which check that for any Integer.toString argument, both functions do not throw and return a user with the given "name".

Alas, that does not really reproduce the initial problem. It doesn't even hit the same code path. The initial problem was to lookup a user/group name for a UID/GID, respectively. For that matter we would need to exercise the `getUsername` and `getGroupName` methods which are only called if the `NativeUserPrincipal`'s `name` is `None`.

Is it possible to access the classes in the `scala.scalanative.nio.fs.attribute` package from the unit-tests?

Reply:
@avdv 
The `Very important notice about Javalib` of the [Contributing Guidelines](https://scala-native.readthedocs.io/en/v0.3.9-docs/contrib/contributing.html#very-important-notice-about-javalibl) say that we can not
use or look at OpenJDk code, so we can not use either the code nor the knowledge of the loophole.

However, we can use knowledge from before looking at the code.  A careful reading of the Java * docs
indicates that all Principals have a name.  The information you provided in the base note indicates that
the name used for gid when there is no other in /etc/groups is gid.

I checked the discontinued Apache Harmony javalib implementation, which we _can_ look at. Its Principals
all throw if a name is not provide.

I think the concept you have surfaced that all Principals have a name allows a major simplification of the code,

Reply:
yes, you are correct, with this generation of changes, the test to prove the base issue was fixed I proposed
is no longer useful.  I'll try again.